modules/sitemap/index.js

```diff
@@ -0,0 +1,46 @@
+const sm = require('sitemap')
+
+module.exports = async function nuxtSitemap (options) {
+
+  // Defaults
+  const defaults = {
+    path: '/sitemap.xml',
+    hostname: null,
+    extendRoutes: [],
+    routes: []
+  }
+
+  // Combine sources
+  const sitemap = Object.assign({}, defaults, this.options.sitemap, options)
+
+  // Extend routes
+  this.extendRoutes((routes, resolve) => {
+    sitemap.extendRoutes = routes.map((route) => route.path)
+  })
+
+  // Server Middleware
+  this.addServerMiddleware({
+    path: sitemap.path,
+    handler (req, res, next) {
+
+      let sitemapConfig = {}
+
+      // Set sitemap hostname
+      if (!sitemap.hostname) {
+        const protocol = req.headers['x-forwarded-proto'] || (req.connection.encrypted ? 'https' : 'http')
+        sitemapConfig.hostname = `${protocol}://${req.headers.host}/`
+      } else {
+        sitemapConfig.hostname = sitemap.hostname
+      }
+
+      // Set sitemap urls
+      sitemapConfig.urls = sitemap.extendRoutes.concat(sitemap.routes)
```

Comment:
`sitemap.extendRoutes` is static (set on nuxt build by  `extendRoute()`)
But `sitemap.routes` is dynamic and it can be an **array** or a **promise** or a **callbcack** (similare to the [generate.routes](https://nuxtjs.org/api/configuration-generate/#routes) ), so I'm not sure that I can set the `sitemapConfig.urls` from outside the handler.